website/server/controllers/api-v3/auth.js

```diff
@@ -130,6 +130,14 @@ api.registerLocal = {
       newUser.registeredThrough = req.headers['x-client']; // Not saved, used to create the correct tasks based on the device used
     }
 
+    // A/B Test 2016-09-12: Start with Sound Enabled?
+    if (Math.random() < 0.5) {
+      newUser.preferences.sound = 'rosstavoTheme';
+      newUser._ABtest = '20160912-soundEnabled';
+    } else {
+      newUser._ABtest = '20160912-soundDisabled';
+    }
+
```

Comment:
I think for now this is fine. If we decide to have multiple tests at the
same time or with more options we can expand it later. Let's not over
engineer

Il 11 set 2016 17:47, "Kaitlin Hipkin" <notifications@github.com> ha
scritto:

> In website/server/controllers/api-v3/auth.js
> <https://github.com/HabitRPG/habitrpg/pull/7984#discussion_r78297414>:
>
> > @@ -130,6 +130,14 @@ api.registerLocal = {
> >        newUser.registeredThrough = req.headers['x-client']; // Not saved, used to create the correct tasks based on the device used
> >      }
> >
> > +    // A/B Test 2016-09-12: Start with Sound Enabled?
> > +    if (Math.random() < 0.5) {
> > +      newUser.preferences.sound = 'rosstavoTheme';
> > +      newUser._ABtest = '20160912-soundEnabled';
> > +    } else {
> > +      newUser._ABtest = '20160912-soundDisabled';
> > +    }
> > +
>
> Suggestions:
>
>    - Move the ABtest logic out of here.
>    - Have some sort of collection of possible ABtests (right now, it
>    would only have this one), where each test has a set of possible options.
>    - Choose one test at random, then choose one option at random.
>
> That way, this logic would allow for multiple ABtests at once, and the
> logic would be modular, hidden away elsewhere.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/HabitRPG/habitrpg/pull/7984/files/513df92ec1b7ed3badea0eedb4657feeebd7dadf#r78297414>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAkAV7aJAnYhyflH3KI2KJM42-vqW71oks5qpCKCgaJpZM4J4b0S>
> .
>